EventFilter/GEMRawToDigi/plugins/GEMDigiToRawModule.cc

```diff
@@ -0,0 +1,262 @@
+      uint16_t amcId = amc;
+
+      for (auto geb : m_gemROMap->getAMC2GEBs(amcId)){
+	
+	uint16_t gebId = geb;
+	uint32_t chamberId = (amcId << 5) | gebId;
+	
+	GEMDetId chamDetId = m_gemROMap->gebPosition(chamberId);
+
+	GEBdata * gebData = new GEBdata();
+	gebData->setInputID(gebId);
+	
+	// 1 GEB per chamber
+	// making vfats
+	for (uint16_t bc = 0; bc < 50; ++bc){
+	  std::map<int, std::vector<int> > vFatToStripMap;
+	  bool hasDigi = false;
+	
+	  for (int roll = 1; roll<=8; ++roll){
+	  
+	    GEMDetId gemId(chamDetId.region(), 1, chamDetId.station(), chamDetId.layer(), chamDetId.chamber(), roll);
+	  
+	    GEMDigiCollection::Range range = gemDigis->get(gemId);
+	    for (GEMDigiCollection::const_iterator digiIt = range.first; digiIt!=range.second; ++digiIt){
+
+	      const GEMDigi & digi = (*digiIt);
+	      if (digi.bx() != bc-25) continue;
+	
+	      GEMROmap::dCoord dc;
+	      dc.gemDetId = gemId;
+	      dc.stripId = digi.strip();
+
+	      GEMROmap::eCoord ec = m_gemROMap->hitPosition(dc);
+	      uint32_t vFatID = ec.vfatId;
+	      int channelId = ec.channelId;
+		
+	      vFatToStripMap[vFatID].push_back(channelId);	
+	      hasDigi = true;
+
+	      // std::cout <<"GEMDigiToRawModule vfatId "<<ec.vfatId
+	      // 		<<" gemDetId "<< gemId
+	      // 		<<" chan "<< ec.channelId
+	      // 		<<" strip "<< dc.stripId
+	      // 		<<" bx "<< digi.bx()
+	      // 		<<std::endl;
+	    }
+	  }
+	  
+	  if (!hasDigi) continue;
+	  
+	  // fill in vFat
+	  std::map<int, std::vector<int> >::const_iterator vFatStrIt = vFatToStripMap.begin();    
+	  for (; vFatStrIt != vFatToStripMap.end(); ++vFatStrIt) {
+
+	    if (vFatStrIt->second.empty()) continue;
+      
+	    uint8_t  b1010      =0xA;           ///<1010:4 Control bits, shoud be 1010
+	    uint16_t BC         =bc;             ///<Bunch Crossing number, 12 bits
+	    uint8_t  b1100      =0xC;           ///<1100:4, Control bits, shoud be 1100
+	    uint8_t  EC         =0;             ///<Event Counter, 8 bits
+	    uint8_t  Flag       =0;             ///<Control Flags: 4 bits, Hamming Error/AFULL/SEUlogic/SUEI2C
+	    uint8_t  b1110      =0xE;           ///<1110:4 Control bits, shoud be 1110
+	    uint16_t crc        =0;             ///<Check Sum value, 16 bits
+	    uint16_t crc_calc   =0;             ///<Check Sum value recalculated, 16 bits
+	    int      SlotNumber =0;             ///<Calculated chip position
+	    bool     isBlockGood=false;         ///<Shows if block is good (control bits, chip ID and CRC checks)
+
+	    uint16_t ChipID = 0xFFF & vFatStrIt->first; ///<Chip ID, 12 bits
+	    uint64_t lsData     =0;             ///<channels from 1to64 
+	    uint64_t msData     =0;             ///<channels from 65to128
+	  
+	    for (auto chan : vFatStrIt->second){
+	      uint64_t oneBit = 0x1;
+	      if (chan < 64) lsData = lsData | (oneBit << chan);
+	      else msData = msData | (oneBit << (chan-64));
+	    }
+
+	    // uint16_t crc = checkCRC(b1010, BC, b1100,
```

Comment:
Is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant